### PR TITLE
Python scalability banks

### PR DESCRIPTION
**Pull request by Soroosh129**
This PR is a supplement to #843. This updates the PythonGenerator to add the following:

- use of the new and updated banks support introduced in #843
    - This has a user-facing side effect that bank index is now also a proper parameter in the Python target that can be passed down the hierarchy (added to RELEASES.md)
- tests introduced in #843 for the C target ported to Python
- support for sending to and receiving from contained banks (necessary for the added tests to pass)
- fixes for formatting issues in `PythonGenerator.xtend` (to make it easier to port to Java)
- changes introduced in #826 (already reviewed)
- updates to use the new CodeBuilder introduced in #892 (needed to fix merge conflicts with `master`)

Most of the changes are due to the format fixes, and are purely superficial.

### CODE REVIEW

**Comment by lhstrh** [comment]
Looks like we should change the base of this to `master`?

**Review by lhstrh — APPROVED** [review]
This looks great. No comments from my side.

File: org.lflang/src/org/lflang/generator/python/PythonGenerator.xtend [hunk 1]
```diff
@@ -1808,17 +1891,84 @@ class PythonGenerator extends CGenerator {
...
     }
     
+    /**
+     * Generate code that creates a Python list (i.e., []) for contained banks to be passed to Python reactions.
+     * The Python reaction will then subsequently be able to address each individual bank member of the contained 
+     * bank using an index or an iterator. Each list member will contain the given <code>port<code> 
+     * (which could be a multiport with a width determined by <code>widthSpec<code>).
+     * 
+     * This is to accommodate reactions like <code>reaction() -> s.out<code> where s is a bank. In this example,
+     * the generate Python function will have the signature <code>reaction_function_0(self, s_out)<code>, where
```
Comment (Soroosh129) [inline org.lflang/src/org/lflang/generator/python/PythonGenerator.xtend:1924]:
```suggestion
     * the generated Python function will have the signature <code>reaction_function_0(self, s_out)<code>, where
```

File: org.lflang/src/org/lflang/generator/python/PythonGenerator.xtend [hunk 2]
```diff
@@ -448,43 +463,64 @@ class PythonGenerator extends CGenerator {
 
     }
     
+    /**
+     * Generate into the specified string builder (<code>inits<code>) the code to
+     * initialize local variable for <code>port<code> so that it can be used in the body of
+     * the Python function.
```
Comment (Soroosh129) [inline org.lflang/src/org/lflang/generator/python/PythonGenerator.xtend:469]:
```suggestion
     * the Python reaction.
```

File: org.lflang/src/org/lflang/generator/python/PythonGenerator.xtend [hunk 3]
```diff
@@ -1,29 +1,28 @@
 /* Generator for the Python target. */
 
 /*************
-Copyright (c) 2019, The University of California at Berkeley.
+ * Copyright (c) 2019, The University of California at Berkeley.
```
Comment (Soroosh129) [inline org.lflang/src/org/lflang/generator/python/PythonGenerator.xtend:4]:
```suggestion
 * Copyright (c) 2022, The University of California at Berkeley.
 * Copyright (c) 2022, The University of Texas at Dallas.
```

File: org.lflang/src/org/lflang/validation/LFValidator.xtend [hunk 4]
```diff
@@ -1287,11 +1288,11 @@ class LFValidator extends BaseLFValidator {
                     Literals.TYPE__STARS
                 )
             }
-        }
-        else if (this.target == Target.Python) {
+        } else if (this.target == Target.Python) {
             if (type !== null) {
                 error(
-                    "Types are not allowed in the Python target",
+                    "Types are not allowed in the Python target (found type " + type.id +
```
Comment (lhstrh) [inline org.lflang/src/org/lflang/validation/LFValidator.xtend:1294]: @housengw: please note this change and incorporate it in your port to Java.

Comment (housengw) [reply org.lflang/src/org/lflang/validation/LFValidator.xtend:1294]: incorporated changes in commit f9741cbc6279d3d267d84f1cdd0f50c497d65749

**Pull request merged**